numbat/src/typed_ast.rs

```diff
@@ -40,13 +40,33 @@ impl DType {
             }
         }
     }
+    /// Is the current dimension type the Time dimension?
+    ///
+    /// This ia special helper that's useful when dealing with DateTimes
+    pub fn is_time_dimension(&self, registry: &DimensionRegistry) -> bool {
+        let mut names = vec![];
+
+        let factors: Vec<_> = self.iter().collect();
+        if factors.len() == 1 && factors[0].1 == Exponent::from_integer(1) {
+            names.push(factors[0].0.clone());
+        }
+
+        names.extend(registry.get_derived_entry_names_for(self));
+
+        match &names[..] {
+            [] => false,
+            [single] if single == "Time" => true,
+            _ => false,
+        }
```

Comment:
It's probably for a reason, but this seems a bit convoluted/complicated. I would have thought something like `self == BaseRepresentation::from_factor(BaseRepresentationFactor("Time".into(), Rational:::from_integer(1)))` might work here?

Reply:
No good reason for this.  I just copy/pasted this code from `to_readable_type()` function, without fully understanding it.  I'll try your suggestion